### PR TITLE
Download given kubeone version and use to provision the cluster

### LINKED ISSUE #717: Make upgrade E2E tests use stable version to provision the cluster
Upgrade E2E tests are using KubeOne built from `master` to provision and then upgrade the cluster. This path doesn't test upgrading components, such as CNI and `machine-controller`, because we always deploy the latest versions when provisioning the cluster.

To increase the coverage and improve tests, we should use the latest stable version of KubeOne to provision the cluster, and then build KubeOne from `master` and run `kubeone upgrade`.

/assign

### PR DESCRIPTION
**Pull request by kron4eg**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #717

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Download given kubeone version and use to provision the cluster
```

### CODE REVIEW

File: testv2/e2e/helpers.go [hunk 1]
```diff
@@ -86,22 +94,111 @@ func requiredTemplateFunc(warn string, input interface{}) (interface{}, error) {
...
+
+	k1Bin, err := os.OpenFile(binPath, os.O_CREATE|os.O_WRONLY, 0750)
+	if err != nil {
+		t.Fatalf("open kubeone destination file: %v", err)
+	}
+	defer k1Bin.Close()
+
+	_, err = io.Copy(k1Bin, unzipK1Bin)
+	if err != nil {
+		t.Fatalf("extractive kubeone from zip: %v", err)
```
Comment (ahmedwaleedmalik) [inline testv2/e2e/helpers.go:168]:
```suggestion
		t.Fatalf("extracting kubeone from zip: %v", err)
```

`extracting or copying`

File: testv2/e2e/scenario_upgrade.go [hunk 2]
```diff
@@ -25,6 +25,8 @@ import (
 	"sigs.k8s.io/yaml"
 )
 
+const kubeoneVersionToInit = "1.4.3"
```
Comment (ahmedwaleedmalik) [inline testv2/e2e/scenario_upgrade.go:28]: Wouldn't it be better to get the latest release dynamically? I mean even here you'll have to update this to `1.4.4`.

File: testv2/e2e/scenario_upgrade.go [hunk 3]
```diff
@@ -43,13 +45,16 @@ func (scenario *scenarioUpgrade) SetVersions(versions ...string) {
 }
 
 func (scenario *scenarioUpgrade) Run(t *testing.T) {
-	t.Helper()
+	if err := makeBin("build").Run(); err != nil {
+		t.Fatalf("building kubeone: %v", err)
```
Comment (ahmedwaleedmalik) [inline testv2/e2e/scenario_upgrade.go:49]: > building kubeone: %v

Failure messages usually have a negative connotation, this should rather be `failed building kubeone binary: %v` or something similar.

Comment (ahmedwaleedmalik) [reply testv2/e2e/scenario_upgrade.go:49]: If you agree with that then most of the `t.Fatalf` blocks need ammendments.

Comment (kron4eg) [reply testv2/e2e/scenario_upgrade.go:49]: No, I don't agree, recently I've rebuild whole errors handling in kubeone and messaging to this form `"doing something: %v", err` which turns into `"doing something: process exited with error 1"` or similar.

File: testv2/e2e/helpers.go (hunk 1 above)
Comment (kron4eg) [reply testv2/e2e/helpers.go]: Fixed :+1:

Comment (kron4eg) [reply testv2/e2e/helpers.go]: Nice catch, even spell checker been confused :D

File: testv2/e2e/scenario_upgrade.go (hunk 2 above)
Comment (ahmedwaleedmalik) [reply testv2/e2e/scenario_upgrade.go:28]: w.r.t https://github.com/kubermatic/kubeone/blob/master/install.sh#L24

Comment (ahmedwaleedmalik) [reply testv2/e2e/scenario_upgrade.go:28]: Hmm, that's something I'll be worried about in this case as well. For now, let's just stick to this hard coding. But would you please update this to `1.4.4`? ;)

Comment (kron4eg) [reply testv2/e2e/scenario_upgrade.go]: done

Comment (ahmedwaleedmalik) [reply testv2/e2e/scenario_upgrade.go]: Thanks

**Review by ahmedwaleedmalik — APPROVED** [review]
/approve ⚡

**Pull request merged**